src/main/java/com/parallels/desktopcloud/ParallelsDesktopConnectorSlaveComputer.java

```diff
@@ -411,4 +414,34 @@ public boolean hasPermission(Permission permission)
 			return false;
 		return super.hasPermission(permission);
 	}
+
+	@Override
+	public void setTemporarilyOffline(boolean temporarilyOffline, OfflineCause cause)
+	{
+		try
+		{
+			Jenkins jenkins = Jenkins.getInstance();
+			for (ParallelsDesktopVM vm : this.getNode().getOwner().getVms())
+			{
+				Node slaveNode = jenkins.getNode(vm.getSlaveName());
+				if (slaveNode == null)
+				{
+					continue;
+				}
+				Computer slaveComputer = slaveNode.toComputer();
+				if (!slaveComputer.isOffline())
```

Comment:
Removed.

What I don't like about this implementation is that having the host and one of its VM slaves connected and offline, I still can bring the VM slave online and jobs will run on it despite the host is offline. I can override VMSlaveComputer.setTemporarilyOffline() to call the connector's method so the host and all the VM slaves will be put offline. But I don't think it's good (because it's not intuitive). Ideally I would prohibit putting VM slaves online/offline from the GUI, but I cannot find an API for it.